Homework 4/Phone book/Phone book/Phone book.c

```diff
@@ -0,0 +1,275 @@
+            {
+                int j = 0;
+                char name[20] = { '\0' };
+                while (buffer[j] != ' ')
+                {
+                    name[j] = buffer[j];
+                    ++j;
+                }
+                ++j;
+                char phone[15] = { '\0' };
+                int counterForPhone = 0;
+                while (buffer[j] != '\0')
+                {
+                    phone[counterForPhone] = buffer[j];
+                    ++j;
+                    ++counterForPhone;
+                }
+                strcpy(array[i].name, name);
+                strcpy(array[i].phone, phone);
+                ++i;
+            }
+        }
+    }
+    return i;
+}
+
+void checkMakingEntryInFile(int indexOfEntries, PhoneBookEntry array[], FILE* test, char* result)
+{
+    int const entriesInFile = indexOfEntries;
+    newEntry(test, indexOfEntries, array, "Kirill", "20202018");
+    newEntry(test, indexOfEntries + 1, array, "Sasha", "890055555");
+    saveInFile(test, entriesInFile, indexOfEntries + 2, array);
+    fclose(test);
+    test = fopen("Test.txt", "r");
+    bool checkOfEqual = true;
+    char resultOfName[20] = { '\0' };
+    char resultOfPhoneNumber[15] = { '\0' };
+    for (int i = 0; i < indexOfEntries; ++i)
+    {
+        fscanf(test, "%s %s", resultOfName, resultOfPhoneNumber);
+    }
+    for (int j = indexOfEntries; j < indexOfEntries + 2; ++j)
+    {
+        fscanf(test, "%s %s", resultOfName, resultOfPhoneNumber);
+        if (strcmp(array[j].name, resultOfName) != 0)
+        {
+            checkOfEqual = false;
+            break;
+        }
+        if (strcmp(array[j].phone, resultOfPhoneNumber) != 0)
+        {
+            checkOfEqual = false;
+            break;
+        }
+    }
+    if (checkOfEqual)
+    {
+        strcpy(result, "ok");
+    }
+    else
+    {
+        strcpy(result, "not ok");
+    }
+}
+
+bool checkPhoneBook(int command, char* expectedResult, char* nameOrNumber)
+{
+    FILE* test = fopen("Test.txt", "a+");
+    if (test == NULL)
+    {
+        printf("File for tests hasn't been found and couldn't be made");
+        return -1;
+    }
+    PhoneBookEntry array[20] = { '\0' };
+    int indexOfEntries = readEntriesFromFile(test, array);
+    char result[20] = { '\0' };
+    if (command == 3)
+    {
+        int const requiredIndex = findPhoneNumber(test, indexOfEntries, array, nameOrNumber);
+        if (requiredIndex != -1)
+        {
+            strcpy(result, array[requiredIndex].phone);
+        }
+    }
+    if (command == 4)
+    {
+        int const requiredIndex = findName(test, indexOfEntries, array, nameOrNumber);
+        if (requiredIndex != -1)
+        {
+            strcpy(result, array[requiredIndex].name);
+        }
+    }
+    if (command == 5)
+    {
+        checkMakingEntryInFile(indexOfEntries, array, test, result);
+    }
+    fclose(test);
+    return(strncmp(expectedResult, result, strlen(expectedResult)) == 0);
```

Comment:
```suggestion
    return strncmp(expectedResult, result, strlen(expectedResult)) == 0;
```
return же не функция